metrics/metrics.go

```diff
@@ -0,0 +1,275 @@
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"github.com/bnb-chain/greenfield-challenger/config"
+)
+
+const (
+	// Monitor
+	MetricGnfdSavedBlock = "gnfd_saved_block"
+	MetricGnfdSavedEvent = "gnfd_saved_event"
+	// increase(gnfd_saved_block[1m])
+	// increase(gnfd_saved_event[1m]) fix this
+
+	// Verifier
+	MetricVerifiedChallenges   = "verified_challenges"
+	MetricChallengeFailed      = "challenge_failed"
+	MetricChallengeSuccess     = "challenge_success"
+	MetricHashVerifierErr      = "hash_verifier_error_count"
+	MetricHashVerifierDuration = "hash_verifier_duration"
+	// increase(verified_challenges[1m])
+	// increase(challenge_success[1m])
+	// increase(challenge_failed[1m])
+
+	// Vote Broadcaster
+	MetricBroadcastedChallenges = "broadcasted_challenges"
+	MetricBroadcasterDuration   = "broadcaster_duration"
+	MetricBroadcasterErr        = "broadcaster_error_count"
+	// increase(broadcasted_challenges[3h]) check increase for heartbeat interval
+
+	// Vote Collector
+	// TODO: Do we need to check how many votes collected per challengeId?
+	// TODO: Should error be a counter or challengeId?
+	// Compare duration monitor/verifier verifier/broadcaster broadcaster/collator collator/submitter
+
+	// Vote Collator
+	MetricCollatedChallenges = "collated_challenges"
```

Comment:
Can we also add some metrics for vote collection?

Reply:
added